File: batches_csharp/batch_11_partial_copy/ref.cs
using System;
using System.Collections.Generic;
using System.IO;

class ProcessData {
    static List<string> FetchLines(string path) => new List<string>(File.ReadAllLines(path));

    static void SaveResults(List<string> items, string path) => File.WriteAllLines(path, items);

    static int Process() {
        var data = FetchLines("input.txt");
        var filtered = new List<string>();
        foreach (var x in data) if (x.Trim().Length > 5) filtered.Add(x.Trim());
        SaveResults(filtered, "output.txt");
        return filtered.Count;
    }

    static void Main() => Console.WriteLine(Process());
}

File: batches_javascript/batch_11_partial_copy/ref.js
const fs = require('fs');

function fetchLines(path) {
    return fs.readFileSync(path, 'utf8').split('\n');
}

function saveResults(items, path) {
    fs.writeFileSync(path, items.join('\n'));
}

function processData() {
    const data = fetchLines('input.txt');
    const filtered = data.filter(x => x.trim().length > 5).map(x => x.trim());
    saveResults(filtered, 'output.txt');
    return filtered.length;
}

console.log(processData());
